content/guides/gha.md

```diff
@@ -0,0 +1,277 @@
+your image using the `docker/build-push-action` by adding the following job:
+
+```yaml
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Test the app
+        uses: docker/build-push-action@v6
+        with:
+          target: tests
+```
+
+This configuration assumes that your Dockerfile contains a stage named `tests`
+that executes your unit tests. The `docker/build-push-action` builds the
+`tests` stage, and if the tests fail during the build, the action fails, and
+the workflow stops.
+
+Since the action uses the Git context by default, you don't need to check out
+the repository explicitly. The `docker/build-push-action` [automatically detects](https://github.com/docker/build-push-action?tab=readme-ov-file#git-context)
+the Git reference of the GitHub Actions workflow and builds with that
+repository as its context.
+
+## Extract metadata for tags and annotations
+
+To automate the tagging and labeling of your Docker images, use the
+`docker/metadata-action`. This action extracts metadata from your Git
+repository, such as branch names and commit SHAs, and uses them to generate
+tags and labels for your images.
+
+Add the metadata action to your workflow:
+
+```yaml
+- name: Extract Docker image metadata
+  id: meta
+  uses: docker/metadata-action@v5
+  with:
+    images: my-dockerhub-user/my-image
+```
+
+> [!NOTE]
+> Replace `my-dockerhub-user/my-image` in the `images` field with your image
+> repository name.
+
+This step prepares the metadata for use in tagging and labeling your images
+during the build and push process.
+
+- The `images` input defines the image repository and name to use for tags and
+  annotations.
+- The `id` property assigns a referable ID for use in later steps to build with
+  the extracted metadata values.
+
+## Authenticate to your registry
+
+Before you build the image, you'll want to authenticate to your registry.
+Authentication is required to push your built image to the registry.
+
+To authenticate with Docker Hub, add the following step to your workflow.
+
+```yaml
+- name: Log in to Docker Hub
+  uses: docker/login-action@v3
+  with:
+    username: ${{ vars.DOCKER_USERNAME }}
+    password: ${{ secrets.DOCKER_PASSWORD }}
+```
+
+This assumes `DOCKER_USERNAME` is configured as a
+[variable](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables),
+and `DOCKER_PASSWORD` as a
+[secret](https://docs.github.com/en/actions/security-for-github-actions/security-guides/using-secrets-in-github-actions),
+in the GitHub repository, containing the username and password for your Docker
+account.
+
+## Build and push the image
+
+Finally, build the final production image and push it to your registry. This
+example also configures the action to generate SBOM and provenance
+[attestations](/manuals/build/metadata/attestations/_index.md) at this stage.
+SBOM (Software Bill of Materials) provides a detailed list of components
```

Comment:
nit: SBOM is defined here, but there's a earlier occurrence in the paragraph. It's also already defined at the beginning of the topic, so maybe not necessary here.